src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -522,23 +566,21 @@ public ElectionInfo(int setVersion, ElectionId electionId)
 
             public ElectionId ElectionId => _electionId;
 
+            public bool IsFresh(int setVersion, ElectionId electionId)
+            {
+                return _setVersion > setVersion
+                       || _setVersion == setVersion && _electionId != null && _electionId.CompareTo(electionId) > 0;
+            }
+            
+            public bool IsSetVersionStale(int setVersion)
+            {
+                return _setVersion < setVersion;
+            }
+            
             public bool IsStale(int setVersion, ElectionId electionId)
             {
-                if (_setVersion < setVersion)
-                {
-                    return true;
-                }
-                if (_setVersion > setVersion)
-                {
-                    return false;
-                }
-
-                if (_electionId == null)
-                {
-                    return true;
-                }
-
-                return _electionId.CompareTo(electionId) <= 0;
+                return IsSetVersionStale(setVersion) 
+                       || _setVersion == setVersion && (_electionId == null || _electionId.CompareTo(electionId) < 0);
```

Comment:
Minor refactoring, but the logic remains the same _except_ that we've changed `<= 0` to `< 0` because we no longer consider equal (`.CompareTo(...) == 0`) ElectionIds to be stale, in order to prevent superfluous log spam as previously discussed.